DataDog/datadog-operator
# Add Features.LogCollection

### What does this PR do?

* Add LogCollection in `spec.features`.
* Add Patch step in DatadogAgent reconcile loop.
* Fixed unittest around orchestrator explorer.

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

* Create a `DatadogAgent` with `spec.agent.log.enabled: true`.
  After a reconcile loop, the `DatadogAgent.spec.features.logCollection.enabled:true`
  should be added.
* Create a `DatadogAgent` with `spec.features.logCollection.enabled:true`.
  Log feature should be enabled in the agent.
